ansible/roles/polkadot-validator/templates/polkadot.service.j2

```diff
@@ -1,47 +1,71 @@
+{# --- PREPARE VARIABLES #}
+{% set nodeName = hostvars[inventory_hostname].nodeName|default(None) %}
+{% set execution = hostvars[inventory_hostname].execution|default(None) %}
+{% set wasmExecution = hostvars[inventory_hostname].wasmExecution|default(None) %}
+{% set telemetryUrl = hostvars[inventory_hostname].telemetryUrl|default(None) %}
+{% set enableReverseProxy = hostvars[inventory_hostname].enableReverseProxy|default(None) %}
+{% set loggingFilter = hostvars[inventory_hostname].loggingFilter|default(None) %}
+{% set basePath = hostvars[inventory_hostname].basePath|default(None) %}
+{% set additionalFlags = hostvars[inventory_hostname].additionalFlags|default(None) %}
+{# ## Legacy/deprecated flags #}
+{% if wasmExecution is none %}
+  {% set wasmExecution = hostvars[inventory_hostname].wasm_execution|default(None) %}
+{% endif %}
+{% if basePath is none %}
+  {% set basePath = hostvars[inventory_hostname].base_path|default(None) %}
+{% endif %}
+{% set additionalCommonFlags = hostvars[inventory_hostname].polkadot_additional_common_flags|default(None) %}
+{% set additionalValidatorFlags = hostvars[inventory_hostname].polkadot_additional_validator_flags|default(None) %}
+{# --- #}
 [Unit]
 Description=Polkadot Node
 
 [Service]
 User=polkadot
 Group=polkadot
 ExecStart=/usr/local/bin/polkadot \
-  {% if hostvars[inventory_hostname].nodeName is defined and hostvars[inventory_hostname].nodeName|length %}
-  --name {{ hostvars[inventory_hostname].nodeName }} \
+  {% if execution is not none and execution|length %}
+  --execution {{ execution }} \
+  {% endif %}
+  {% if nodeName is not none and nodeName|length %}
+  --name {{ nodeName }} \
   {% else %}
   --name {{ project|default('project') }}-sv-validator-{{ groups['validator'].index(inventory_hostname) }} \
   {% endif %}
-  --validator \
-  {% if hostvars[inventory_hostname].enableReverseProxy is defined and hostvars[inventory_hostname].enableReverseProxy %}
-  --public-addr=/ip4/{{ hostvars[inventory_hostname].public_ip.json.ip }}/tcp/{{ proxy_port }} \
+  {% if wasmExecution is not none and wasmExecution|length %}
+  --wasm-execution {{ wasmExecution }} \
+  {% else %}
+  --wasm-execution Compiled \
```

Comment:
probably not needed as of 0.9.3 as it's the default setting

Reply:
Sure, but I'll leave it there anyway since it might be useful in general, for example when there's a rollback as experienced recently.